torch/_inductor/lowering.py

```diff
@@ -1385,16 +1390,16 @@ def inner_fn(idx):
         scale = scales_loader(channel_idx)
         zero_point = zero_points_loader(channel_idx)
 
-        if scales.dtype != torch.float32:
-            scale = ops.to_dtype(scale, torch.float32)
-        if zero_points.dtype != torch.float32:
-            zero_point = ops.to_dtype(zero_point, torch.float32)
-        val = ops.sub(ops.to_dtype(input, torch.float32), zero_point) * scale
+        if scales.dtype != out_dtype:
+            scale = ops.to_dtype(scale, out_dtype)
+        if zero_points.dtype != out_dtype:
+            zero_point = ops.to_dtype(zero_point, out_dtype)
+        val = ops.sub(ops.to_dtype(input, out_dtype), zero_point) * scale
```

Comment:
Can we only cast the output to out_dtype? it may saves unnecessary cast I guess.

Reply:
should scales/zps dtype always match out dtype?

Reply:
We decide to keep the original code, to cast scale/zp/input to `torch.float32` before calculations of `sub` and `mul`. Finally, we cast the `val` to `out_dtype`.

Reply:
I have updated the PR, please help review again.